platform/android/java_godot_view_wrapper.cpp

```diff
@@ -0,0 +1,54 @@
+/*************************************************************************/
+/*  java_godot_view_wrapper.cpp                                          */
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2020 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2020 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "java_godot_view_wrapper.h"
```

Comment:
Instead of creating a new jni wrapper class, you can add proxy methods in `Godot.java` and call them from `java_godot_wrapper`. This will also allow you to validate whether the methods are available since they were only introduced in API level 26, and we support as far back as API level 18.

Reply:
Sorry, I don't understand how does it help to add that directly to godot wrapper help, doesn't adding another wrapper make it cleaner and easier to check, can you elaborate?
Also I can add check to see if we run on new enough Android version and do nothing if not

Reply:
Sure, [`requestPointerCapture()`](https://developer.android.com/reference/android/view/View#requestPointerCapture()) and [`releasePointerCapture()`](https://developer.android.com/reference/android/view/View#releasePointerCapture()) were introduced in API level 26. Since we support API level 18 and up, on devices running version < 26, you would get a crash since the methods you're invoking via jni are not available.

To counter that, you can add proxy methods in `Godot.java` as shown below:
```
void requestPointerCapture() {
    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
        mRenderView.requestPointerCapture();
    }
    // Do nothing otherwise.
}

// Same for releasePointerCapture()
```

And to invoke these methods, you'd add jni accessors in `java_godot_wrapper`.

Reply:
I can also just test that in the cpp side (with `android_get_device_api_level()` https://developer.android.com/ndk/reference/group/apilevels#android_get_device_api_level )
And I think it will make the code cleaner, also Ill probably use this class also when Ill add cursor icon support

Reply:
Yes that would work also!

Reply:
Sorry for the delay, working on it currently

will be ready today/tommarow

Reply:
12 days later that I thought, ready